pkg/kubelet/kubelet.go

```diff
@@ -795,6 +802,9 @@ func (kl *Kubelet) initialNodeStatus() (*api.Node, error) {
 			Name:   kl.nodeName,
 			Labels: map[string]string{"kubernetes.io/hostname": kl.hostname},
 		},
+		Spec: api.NodeSpec{
```

Comment:
> Folks running the OSS code may want to provide API access (but not direct machine access) to their cluster to other users (e.g. here's a copy of my kubeconfig but you don't have ssh access to the master). This change means that you can't effectively do that

In OSS today, you can create users with access to only specific resources and specific namespaces with ABAC authorizer. An admin can restrict write access to the node resource.